frontend/online-shop/src/pages/edit-product/edit-product.page.tsx

```diff
@@ -1,29 +1,27 @@
 import { Button, TextField, Typography } from "@mui/material";
-import { useState, useEffect } from "react";
+import { useEffect } from "react";
 import { useNavigate, useParams } from "react-router-dom";
 import { routes } from "../../helpers/routes";
 import { Product } from "../../models/product.model";
+import { updateProductById } from "../../services/products.service";
+import { useAppDispatch, useAppSelector } from "../../store/hooks";
 import {
-  fetchProductById,
-  updateProductById,
-} from "../../services/products.service";
+  fetchEditProductStart,
+  updateProduct,
+} from "../../store/slices/edit-product-slice";
 
 const EditProduct = () => {
   const params = useParams();
   const navigate = useNavigate();
-  const [product, setProduct] = useState<Product>();
-
+  const dispatch = useAppDispatch();
+  const product = useAppSelector((state) => state.editProduct.product);
   useEffect(() => {
-    const fetch = async () => {
-      const res = await fetchProductById(params.id || "");
-      setProduct(res.data);
-    };
-    fetch();
-  }, [params]);
+    dispatch(fetchEditProductStart(params.id || ""));
+  }, [params, dispatch]);
 
   const handleSave = async (product: Product) => {
     await updateProductById(product.id, product);
-    navigate(routes.navigate, { replace: true });
+    navigate(`../products/${product.id}`, { replace: true });
```

Comment:
This should also be a path within the `routes` file. You can have `/products/{productId}` and then replace the mapping with the actual value